### DISCUSSION TITLE
Set/Get loop count of animated webp

### DISCUSSION THREAD

**Discussion by cristalcho** [ImageMagick/ImageMagick#2863] [Development]
Hello, I`m using imagemagick very well and I have one suggestion.

I`m in trouble with changing animated webp image with loop count to gif.
When I checked, setting the loop count for gif is possible with MagickSetImageIterations().
However, MagickGetImageIterations() is not work with animated webp.

Could you improve it to make it possible?
I would be very happy if setting and getting loop count of animated webp works same way as gif.
(with MagickSetImageIterations, MagickGetImageIterations)

Thank you.

**Comment by urban-warrior** [comment]
Does WebP support loops?  We leverage the libwebp delegate library to read and write WebP images.  Does libwebp have a method to get/set the loop (iterations) metadata?  If so, let us know which API call to make and we will add support to ImageMagick for WebP looping.

**Comment by cristalcho** [comment]
I could found WebpAnimEncoder/Decoder API supports loops.
Here is the guides with animated webp api.
I hope this will be helpful. Thank you :)

https://developers.google.com/speed/webp/docs/container-api#webpanimdecoder_api

**Comment by urban-warrior** [comment]
We now can read the loop count from a WebP image.  We're working on setting the loop variable when writing the WebP image format.

**Reply by diego-choi** [reply]
@urban-warrior 

Hello, I'm interested in this discussion. I am also using MagickGetImageIterations / MagickSetImageIterations functions for preserving loop count when I convert gif to webp but it doesn't seem to work MagickSetImageIterations function for webp properly. MagickGetImageIterations works fine for webp as you mentioned. I tested it with ImageMagick-6.9.12-13. I would be happy if setting loop count of animated webp works fine with MagickSetImageIterations. Thank you.

**Reply by diego-choi** [reply]
@urban-warrior 

I found commit (af1b1b66cada4e2832354095e15b0a4a55fdd502) in ImageMagick7 for setting loop count in animated webp but I couldn't found any commit in ImageMagick6, Could you work for ImageMagick6 too?
